### PR TITLE
[16.0][MIG] product_code_mandatory

### PR DESCRIPTION
**Pull request by MarioLM-23**
Improve test classes with **setUpClass**.
Last PR closed of migration module: [#1237](https://github.com/OCA/product-attribute/pull/1237).
Checked two last [issues](https://github.com/OCA/product-attribute/pull/1237#pullrequestreview-1230977029).

### CODE REVIEW

**Comment by mgc-adhoc** [comment]
LGTM Functional +1 @agyamuta @pedrocasi

**Comment by bruno-zanotti** [comment]
Hey @mariobinhex could you squash migration commits?

**Comment by MarioLM-23** [comment]
> Hey @mariobinhex could you squash migration commits?

Done! I have maintained the pre-commit and migration

**Comment by mgc-adhoc** [comment]
@OCA/product-maintainers could you please merge?
Thanks!

**Comment by florenciafrigieri2** [comment]
@rafaelbn Hello Rafal, can you please check this PR?

**Comment by jcadhoc** [comment]
@rafaelbn hi Rafael, can you please check this PR and merge?

**Comment by jcadhoc** [comment]
@simahawk hi Simone, can you please check this PR and merge? Thank you!

File: product_code_mandatory/models/product.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def _get_default_code(self):
+        return self.env["ir.sequence"].next_by_code("product.default.code")
+
+    default_code = fields.Char(
+        "Internal Reference", index=True, default=_get_default_code
```
Comment (simahawk) [inline product_code_mandatory/models/product.py:14]:
```suggestion
        "Internal Reference", index=True, default=lambda self: self._get_default_code()
```
To be done in a separated commit that can be back/fwd ported

File: product_code_mandatory/tests/test_product_code.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import TransactionCase
+
+
+class TestProductCode(TransactionCase):
```
Comment (simahawk) [inline product_code_mandatory/tests/test_product_code.py:6]:
```suggestion
from odoo.tests.common import BaseCommon


class TestProductCode(BaseCommon):
```
to disable tracking

**Review by simahawk — CHANGES_REQUESTED** [review]
Thanks for your contrib.
Please check your history.
The migration commit contains changes to other modules and there's a merge commit that shouldn't be there.

Please, rebase on 16.0 instead and re-apply the migration commits.

**Comment by ced-adhoc** [comment]
Hi @MarioLM-23 
Have you had a chance to address the request on the pr?
Thanks!

**Pull request closed without merging**